### PR TITLE
Add check to async_upload to avoid silent s3cmd failure

### PR DESCRIPTION
**Pull request by lwgordonimos**

### CODE REVIEW

**Comment by lwgordonimos** [comment]
Fixes https://github.com/aodn/chef/issues/366

The s3cmd has *very* inadequate handling of errors, and so returns 0 when at least 2 (very likely more) scenarios occur:
1) the celery user doesn't have read access to the file
...
1000000) the file doesn't exist
etc.

Ideally this will be uploaded using native Python AWS (i.e. boto), but in the mean time this adds a simple test to cause the task to fail and at least log the fact that it has failed.

e.g.

```bash
14-nsp-mel:/var/log/supervisor# s3cmd --config=/mnt/ebs/data-services/s3cfg sync --no-preserve /root/NOFILEHERE s3://imos-test-data/NOFILEHERE
14-nsp-mel:/var/log/supervisor# echo $?
0
```

**Comment by julian1** [comment]
Is there a python S3cmd equivalent?

Would it be better to replace s3cmd wholesale?

**Comment by julian1** [comment]
Also, I believe that [aws-cli](https://aws.amazon.com/cli/) the official cli client that will have AWS support. While s3cmd is a third party app. It would be interesting to know if the error-handling is better with the official client.

**Comment by lwgordonimos** [comment]
@julian1 I briefly referenced boto which is the Python AWS library, which would be preferable to shelling out to anything since it can be done in native Python.

We should definitely change it to either of the more robust tools (cli for bash or boto lib for Python) but at a cursory glance a heck of a lot of things are built around s3cmd specifically so would need substantial testing before migrating, hence this very conservative check in the interim. 

We would need to decide, as a rule, that we consider boto as "safe" to handle file uploads. I suspect it will be fine, since it is the AWS endorsed library (https://aws.amazon.com/sdk-for-python/) but didn't want to make such a change here without it being well tested first and also done with the wider scope in mind.

File: cookbooks/imos_po/templates/default/tasks.py.erb [hunk 1]
```diff
@@ -40,15 +42,17 @@ s3cmd_quoted = s3cmd_parts.map{ |s| "\"#{s}\""}.join(', ')
...
 def async_upload(src_path, dst_path):
+    if not os.access(src_path, os.R_OK):
+        raise Exception("ERROR uploading '{0}'. The file must exist and user '{1}' must have read access to it.".format(src_path, getpass.getuser()))
     command = [
         <%= s3cmd_quoted %>,
         "sync",
         "--no-preserve",
         src_path,
-        path.join("<%= node['imos_po']['s3']['bucket'] %>", dst_path)
+        os.path.join("<%= node['imos_po']['s3']['bucket'] %>", dst_path)
```
Comment (julian1) [inline cookbooks/imos_po/templates/default/tasks.py.erb]: Can't believe we templatized the python to achieve the necessary level of expressibility. And hanging a dependency like our entire chef-framework on this in order to generate runnable code. 

Should be re-factored and made config/data driven.

**Pull request merged**